prov/rxm/src/rxm.h

```diff
@@ -701,28 +707,46 @@ rxm_process_recv_entry(struct rxm_recv_queue *recv_queue,
 		if (rx_buf->pkt.ctrl_hdr.type != ofi_ctrl_seg_data) {
 			return rxm_cq_handle_rx_buf(rx_buf);
 		} else {
+			struct dlist_entry *entry;
 			enum rxm_sar_seg_type last =
 				(rxm_sar_get_seg_type(&rx_buf->pkt.ctrl_hdr)
 								== RXM_SAR_SEG_LAST);
 			ssize_t ret = rxm_cq_handle_rx_buf(rx_buf);
+			struct rxm_recv_match_attr match_attr = {
+				.addr	= recv_entry->addr,
+				.tag	= recv_entry->tag,
+				.ignore	= recv_entry->ignore,
```

Comment:
This initializes match_attr right before we might return from the call.  This would be better moved down past the if check.

Reply:
I agree. Done